flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -588,15 +653,16 @@ protected int cancel(String[] args) {
 			ActorGateway jobManager = getJobManagerGateway(options);
 			Future<Object> response = jobManager.ask(new CancelJob(jobId), askTimeout);
 
-			try {
-				Await.result(response, askTimeout);
-				return 0;
-			}
-			catch (Exception e) {
-				throw new Exception("Canceling the job with ID " + jobId + " failed.", e);
+			Object rc = Await.result(response, askTimeout);
+
+			if (rc instanceof CancellationFailure) {
```

Comment:
Well, I think with that you can wait a little bit because there is an ongoing discussion about explicit messages or not in #1048. I guess we should align according to the result of the discussion.